include/pybind11/gil_safe_call_once.h

```diff
@@ -3,6 +3,7 @@
 #pragma once
 
 #include "detail/common.h"
+#include "detail/type_caster_base.h"
```

Comment:
Sorry I think this is going in the wrong direction:

It brings in a ton of dependencies unrelated to the `gil_safe_call_once` functionality.

It's seems counter-productive to turn on compiler warnings just to appease them.

A great step forward would be to refactor the code organization. I'd be very happy if you did that, even if several pieces of code need to be moved around.

If you think that's too much work: Simply disable the warnings for the troublesome existing forward declaration. That'd be a net gain: Having the warnings enabled will nudge us in the right direction when working on new code. The warning suppressions will remind us that we have a code organization cleanup job waiting for an intrepid volunteer.

Reply:
Fair enough. For restructuring the code, I'm not an expert enough. I will disable the warnings. Switching to draft state for now.